source/UF/SizeTruncationandConnectedness.lagda

```diff
@@ -0,0 +1,346 @@
+                connected
+     cod-con = trunc-map-is-connected ∣ x' ∣[ n + 1 ]
+     f-con : f is n connected-map
+     f-con = basepoint-map-is-less-connected (ua _) f cod-con
+     𝟙-is-small : 𝟙 {𝓤} is 𝓥 small
+     𝟙-is-small = (𝟙 {𝓥} , one-𝟙-only)
+     cod-locally-small : (Σ x ꞉ X , ∣ x ∣[ n + 1 ] ＝ ∣ x' ∣[ n + 1 ]) is
+                          ι (n + 2) locally-small
+     cod-locally-small = locally-small-closed-under-Σ X-loc-small
+                          path-locally-small
+      where
+       path-locally-small : (x : X)
+                          → (∣ x ∣[ succ n ] ＝ ∣ x' ∣[ succ n ]) is
+                           ι (n + 2) locally-small
+       path-locally-small x =
+        locally-small-closed-under-≃ (eliminated-trunc-identity-char (ua _))
+                                     path-char-locally-small
+         where
+          path-char-locally-small : ∥ x ＝ x' ∥[ n ] is ι (n + 2) locally-small
+          path-char-locally-small =
+           locally-small-closed-under-truncation
+            (locally-small-is-upper-closed X-loc-small x x')
+   fiber-path-space-small : (y : ∥ X ∥[ n + 1 ])
+                          → (Σ x ꞉ X , ∣ x ∣[ n + 1 ] ＝ y) is 𝓥 small
+   fiber-path-space-small =
+    ∥∥ₙ-ind (λ - → props-are-truncated pt
+                    (being-small-is-prop ua (Σ x ꞉ X , ∣ x ∣[ n + 1 ] ＝ -) 𝓥))
+            trunc-ind-helper
+
+ small-from-locally-and-trunc-small : {X : 𝓤 ̇} {n : ℕ₋₂}
+                                    → Join-Construction-Result {𝓤} {𝓤}
+                                    → X is ι (n + 2) locally-small
+                                     × ∥ X ∥[ n + 1 ] is 𝓥 small
+                                    → X is 𝓥 small
+ small-from-locally-and-trunc-small {_} {X} {n} j small-hyp =
+  smallness-closed-under-≃' (small-path-space-lemma j small-hyp)
+                            (domain-is-total-fiber ∣_∣[ succ n ])
+
+ Theorem-2-6 : {X : 𝓤 ̇} {n : ℕ₋₂}
+             → Join-Construction-Result {𝓤} {𝓤}
+             → X is 𝓥 small
+             ↔ X is ι (n + 2) locally-small × ∥ X ∥[ n + 1 ] is 𝓥 small 
+ Theorem-2-6 {_} {X} {n} j =
+  (foreward , small-from-locally-and-trunc-small j)
+  where
+   foreward : X is 𝓥 small
+            → X is ι (n + 2) locally-small × ∥ X ∥[ n + 1 ] is 𝓥 small
+   foreward X-small =
+    (locally-small-from-small X-small
+     , truncations-of-small-types-are-small X-small)
+
+\end{code}
+
+Corollary 2.7.
+
+\begin{code}
+
+ Corollary-2-7 : {X : 𝓤 ̇} {Y : 𝓦 ̇} {f : X → Y} {n : ℕ₋₂}
+               → Join-Construction-Result {𝓤} {𝓤}
+               → Propositional-Resizing
+               → f is (n + 1) truncated-map
+               → Y is ι (n + 2) locally-small
+               → ∥ X ∥[ n + 1 ] is 𝓥 small
+               → X is 𝓥 small
+ Corollary-2-7 j pr f-trunc Y-loc-small trunc-X-small =
+  small-from-locally-and-trunc-small j
+   (Lemma-2-4 pr f-trunc Y-loc-small , trunc-X-small)
+
+\end{code}
+
+TODO: Proposition 2.8. requires the concept of Homotopy Groups.
```

Comment:
"TODO.".

Nice! Perhaps in the near future you would like to formalize the join construction.